go/pkg/client/bytestream.go

```diff
@@ -73,6 +73,69 @@ func (c *Client) writeChunked(ctx context.Context, name string, ch *chunker.Chun
 	return totalBytes, err
 }
 
+// WriteBytesWithOffset uploads a byte slice.
+func (c *Client) WriteBytesWithOffset(ctx context.Context, name string, data []byte, opts *ByteStreamWriteOpts) (int64, error) {
+	ue := uploadinfo.EntryFromBlob(data)
+	ch, err := chunker.New(ue, false, int(c.ChunkMaxSize))
+	if err != nil {
+		return 0, err
+	}
+	return c.writeChunkedWithOffset(ctx, name, ch, opts)
+}
+
+// writeChunkedWithOffset uploads chunked data with a given resource name to the CAS at an arbitrary offset.
+func (c *Client) writeChunkedWithOffset(ctx context.Context, name string, ch *chunker.Chunker, opts *ByteStreamWriteOpts) (int64, error) {
+	var totalBytes int64
+	var offset = opts.LastLogicalOffset
+	var initialOffset = opts.LastLogicalOffset
```

Comment:
Doesn't look like you're using the initialOffset in the first write?

Reply:
No it's no being used. I created initialOffset to update offset after writing in line 127.

Reply:
I'm confused. The chunker has chunked the entire blob, right? So its actually disregarding the initial offset? How will you make sure that you're uploading the right chunk relative to the desired offset?

Reply:
Right the current implementation is disregarding the initial offset(offset property in Chunker), I think I can update the code to use the offset property for a clarity. Let me try to update the code.